### PR TITLE
Create restaurant model and schema

### PR DESCRIPTION
**Pull request by jmzxhu**
We added a Restaurant model, schema, API Restaurant controller, set up the routes, Restaurant controller, Restaurant Index.html.erb view as a static page, React components: Restaurant Container, Restaurant Tile,  API controlling testing, unit testing for validation for the models, data to the seed.

### CODE REVIEW

File: app/javascript/react/components/RestaurantIndexContainer.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+return(
+  <div className="grid-container">
+    <div className="tile-container grid-x grid-margin-x">
+        {restaurantArray}
+    </div>
+  </div>
+)
+}
+
```
Comment (nwalberts) [inline app/javascript/react/components/RestaurantIndexContainer.js:41]: really nice work here. If I make more suggestions though, looks like line 39 may need to be indented

File: app/javascript/react/components/RestaurantTile.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+const RestaurantTile = (props) => {
+
+  return (
+    <div className="tile cell small-12 medium-4 large-4">
+      <img className="tile-image" src={props.restaurant.picture_url}/>
+      <div className="tile-text">{props.restaurant.name}</div>
+    </div>
+  )
+}
+export default RestaurantTile
```
Comment (nwalberts) [inline app/javascript/react/components/RestaurantTile.js:12]: styling edit: remove line 4 and add a line below 11

File: app/javascript/react/components/RestaurantIndexContainer.js (hunk 1 above)
Comment (veeveeanne) [reply app/javascript/react/components/RestaurantIndexContainer.js:41]: Fixed this indentation in the most recent PR (#5)

File: app/javascript/react/components/RestaurantTile.js (hunk 2 above)
Comment (veeveeanne) [reply app/javascript/react/components/RestaurantTile.js:12]: Fixed the styling edit in the most recent PR (#5)

**Pull request merged**